### PR TITLE
introduce 'ipc.write' over XHR

### PR DESCRIPTION
**Pull request by jwerle**

### CODE REVIEW

File: util.js [hunk 1]
```diff
@@ -16,6 +16,10 @@ function isFunction (value) {
   return typeof value === 'function' && !/class/.test(value.toString())
 }
 
+function isPromiseLike (object) {
```
Comment (jwerle) [inline util.js:19]: cc @chicoxyzzy

File: node/fs/handle.js [hunk 2]
```diff
@@ -272,8 +272,7 @@ class FileHandle extends EventEmitter {
    * @TODO
```
Comment (chicoxyzzy) [inline node/fs/handle.js:272]: not necessary anymore

File: node/fs/handle.js [hunk 3]
```diff
@@ -301,6 +300,43 @@ class FileHandle extends EventEmitter {
...
```
Comment (chicoxyzzy) [inline node/fs/handle.js:300]: not necessary anymore

**Review by chicoxyzzy — APPROVED** [review]
LGTM!

File: node/fs/handle.js (hunk 3 above)
Comment (jwerle) [reply node/fs/handle.js:300]: I have `TODO` there for me to come input a comment 🙏🏼

**Pull request merged**